internal/handlers/pg/msg_find.go

```diff
@@ -121,22 +123,21 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 			return lazyerrors.Error(err)
 		}
 
-		// TODO https://github.com/FerretDB/FerretDB/issues/1733
-		resDocs, err = iterator.ConsumeValues(iterator.Interface[struct{}, *types.Document](iter))
-
-		return err
+		return nil
 	})
 
 	if err != nil {
-		cancel()
+		closer.Close()
 		return nil, lazyerrors.Error(err)
 	}
 
-	iter := iterator.Values(iterator.ForSlice(resDocs))
-	closer := iterator.NewMultiCloser(iter, iterator.CloserFunc(cancel))
+	closer.Add(iterator.CloserFunc(func() {
+		// ctx could be cancelled already
+		_ = keepTx.Rollback(context.Background())
```

Comment:
What is the reason that we rollback in closer? Maybe because `find` should not modify data? :thinking:

Reply:
It does not matter if we commit transaction or rollback it, but we should close it

Reply:
Got it thanks :+1: